### PR TITLE
Add pkg/operator/render as a generic render command

### PR DESCRIPTION
**Pull request by sttts**
Example usage in https://github.com/openshift/cluster-kube-controller-manager-operator/pull/53.

### CODE REVIEW

File: pkg/operator/render/options/generic.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	// load and render templates
+	if cfg.Assets, err = assets.LoadFilesRecursively(o.AssetInputDir); err != nil {
+		return fmt.Errorf("failed loading assets from %q: %v", o.AssetInputDir, err)
+	}
+
+	return nil
+}
+
+func (o *GenericOptions) configFromDefaultsPlusOverride(templateData interface{}, tlsOverride, bootstrapVersion string, loadAsset LoadAsset) ([]byte, error) {
+	defaultConfig := loadAsset(filepath.Join(bootstrapVersion, "kube-controller-manager", "defaultconfig.yaml"))
```
Comment (deads2k) [inline pkg/operator/render/options/generic.go]: this looks specific, not generic.

File: pkg/operator/render/options/generic.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+// LoadAsset loads asset files.
+type LoadAsset func(name string) []byte
+
+// AddFlags adds the generic flags to the flagset.
+func (o *GenericOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.AssetOutputDir, "asset-output-dir", "", "Output path for rendered manifests.")
+	fs.StringVar(&o.AssetInputDir, "asset-input-dir", "", "A path to directory with certificates and secrets.")
+	fs.StringVar(&o.TemplatesDir, "templates-input-dir", "/usr/share/bootkube/manifests", "A path to a directory with manifest templates.")
```
Comment (deads2k) [inline pkg/operator/render/options/generic.go]: don't do defaults like this.  Default values based on current values and have a `NewOptions` method.

File: pkg/operator/render/options/manifest.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	ImagePullPolicy       string
+	ConfigHostPath        string
+	ConfigFileName        string
+	CloudProviderHostPath string
+	SecretsHostPath       string
+}
+
+// AddfFlags adds the manifest related flags to the flagset.
+func (o *ManifestOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.Namespace, "manifest-namespace", "openshift-kube-controller-manager",
```
Comment (deads2k) [inline pkg/operator/render/options/manifest.go]: same comment about defaults and this looks very specific.

**Comment by deads2k** [comment]
I'm fine with it in concept, but it needs to be generalized and apply lessons about defaults values learned from `oc`.

File: pkg/operator/render/options/generic.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/render/options/generic.go]: indeed, will fix

File: pkg/operator/render/options/manifest.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	ConfigFileName        string
+	CloudProviderHostPath string
+	SecretsHostPath       string
+}
+
+// NewManifestOptions return default values for ManifestOptions.
+func NewManifestOptions(componentName string) *ManifestOptions {
+	return &ManifestOptions{
+		Namespace:             fmt.Sprintf("openshift-%s", componentName)
+		Image:                 "openshift/origin-hyperkube:latest",
```
Comment (deads2k) [inline pkg/operator/render/options/manifest.go]: this image is likely to be different.  An arg seems appropriate.

File: pkg/operator/render/options/manifest.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		ImagePullPolicy:       "IfNotPresent",
```
Comment (deads2k) [inline pkg/operator/render/options/manifest.go]: I've found myself switching these to `Always`, now that I'm not building the source locally like I did for cluster up.

**Comment by deads2k** [comment]
> @deads2k addressed your comments. ptal.

two minor comments and lgtm.

**Comment by deads2k** [comment]
compile error though.

File: pkg/operator/render/options/manifest.go (hunk 4 above)
Comment (sttts) [reply pkg/operator/render/options/manifest.go]: Then let's leave it empty. wdyt?

Comment (sttts) [reply pkg/operator/render/options/manifest.go]: flag is more explicit maybe. Having an image set in the flag makes certainly sense. Switching to func param.

File: pkg/operator/render/options/manifest.go (hunk 5 above)
Comment (sttts) [reply pkg/operator/render/options/manifest.go]: fixed

File: pkg/operator/render/options/manifest.go (hunk 4 above)
Comment (sttts) [reply pkg/operator/render/options/manifest.go]: fixed

**Comment by sttts** [comment]
Also compile error fixed.

**Pull request merged**